### PR TITLE
Code style improvements in JavaScript

### PR DESCRIPTION
**Pull request by berkes**
This fixes errors reported by `yarn run test:lint`.

Three linting errors are not fixed: 
```

xxx/app/javascript/mastodon/actions/announcements.js
  31:3  error  Expected catch() or return  promise/catch-or-return

xxx/app/javascript/mastodon/actions/notifications.js
  152:5  error  Expected catch() or return  promise/catch-or-return

xxx/app/javascript/mastodon/actions/timelines.js
  97:5  error  Expected catch() or return  promise/catch-or-return

✖ 3 problems (3 errors, 0 warnings)
```
I'm not confident enough with the intricacies of our promises here. There are already catch() clauses, but apparently the linter detects other promises that have no catch() clauses. I'm not certain which ones this would be (or if the linter is simply wrong).

### CODE REVIEW

File: app/javascript/mastodon/features/compose/components/poll_form.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ class PollForm extends ImmutablePureComponent {
         <div className='poll__footer'>
           <button disabled={options.size >= 4} className='button button-secondary' onClick={this.handleAddOption}><Icon id='plus' /> <FormattedMessage {...messages.add_option} /></button>
 
-          <select value={expiresIn} onChange={this.handleSelectDuration}>
+          <select value={expiresIn} onBlur={this.handleSelectDuration}>
```
Comment (ThibG) [inline app/javascript/mastodon/features/compose/components/poll_form.js]: This would introduce a bug (see #12596)

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
I'm not convinced by the usefulness of this change, but it's ok save for the part I commented inline.

File: app/javascript/mastodon/features/compose/components/poll_form.js (hunk 1 above)
Comment (berkes) [reply app/javascript/mastodon/features/compose/components/poll_form.js]: I've reverted this, and added a line to configure the linter to ignore this.

**Comment by berkes** [comment]
> I'm not convinced by the usefulness of this change

Currently there is an eslinter configured and set. Either that should go, or it should be honoured, IMO. It makes no sense to have the eslinter wired up, run in tests, but have it fail. That is, arguably, worse than having no linter at all.

Removing the linter would, IMO, be a good alternative if we're not following it anyway. But I'd say, since the amount of errors is small, keeping it in, and keeping running it, is the correct thing to do. 

My plan was to make a PR that enables the js tests in CI, once the linter passes (linting is the first part of running the tests, currently).

**Review by ClearlyClaire — APPROVED** [review]
I'm still somehow half-convinced by the need for that PR, but it looks good to me.

We mostly rely on CodeClimate for coding style, I'm not against removing the linter from the test suite, but I have no strong opinion either way.

**Pull request merged**